lib/fs.js

```diff
@@ -1635,7 +1635,33 @@ function stat(path, options = { bigint: false }, callback) {
   callback = makeStatsCallback(callback);
 
   const req = new FSReqCallback(options.bigint);
-  req.oncomplete = callback;
+
+  options = getOptions(options, { bigint: false, signal: undefined });
+
+  if (options.signal?.aborted) {
+    return process.nextTick(() =>
```

Comment:
Here I would create the error outside the `process.nextTick(...)` so that it has a useful stack trace attached to it.

Reply:
thanks for suggestion 🙏